ios/brave-ios/Sources/Shared/Reachability.swift

```diff
@@ -1,99 +1,121 @@
-// The MIT License (MIT)
-//
-// Copyright (c) 2015 Isuru Nanayakkara
-//
-// Permission is hereby granted, free of charge, to any person obtaining a copy
-// of this software and associated documentation files (the "Software"), to deal
-// in the Software without restriction, including without limitation the rights
-// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
-// copies of the Software, and to permit persons to whom the Software is
-// furnished to do so, subject to the following conditions:
-//
-// The above copyright notice and this permission notice shall be included in all
-// copies or substantial portions of the Software.
-//
-// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-// SOFTWARE.
+// Copyright 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
+import Combine
 import Foundation
-import SystemConfiguration
+import Network
 
 public enum ReachabilityType: CustomStringConvertible {
-  case wwan
-  case wiFi
+  case wifi
+  case cellular
+  case ethernet
+  case other
+  case offline
 
   public var description: String {
     switch self {
-    case .wwan: return "WWAN"
-    case .wiFi: return "WiFi"
+    case .wifi: return "WiFi"
+    case .cellular: return "Cellular"
+    case .ethernet: return "Ethernet"
+    case .other: return "Other"
+    case .offline: return "Offline"
     }
   }
 }
 
-public enum ReachabilityStatus: CustomStringConvertible {
-  case offline
-  case online(ReachabilityType)
-  case unknown
+open class Reachability {
```

Comment:
We have a class called `NetworkMonitor` already that is doing some of the same thing. We probably want to delete it. Since we're also basically changing the entire implementation we should make a new file and put it in BraveShared instead